### PR TITLE
support conv1d pruning

### PR DESCRIPTION
**Pull request by n1ck-guo**
## Type of Change

feature

## Description

support conv1d pruning

## Expected Behavior & Potential Risk

None
## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

None

### CODE REVIEW

**Review by yintong-lu — APPROVED** [review]

File: neural_compressor/compression/pruner/patterns/ninm.py [hunk 1]
```diff
@@ -152,6 +152,9 @@ def _reshape_orig_to_2dims(self, data):
         if len(data.shape) == 4:  # TODO: need to verify whether it's ok for transposed conv
             data = data.permute(0, 2, 3, 1)  # cout,k,k,cin
             data = data.reshape(data.shape[0], -1)
+        if len(data.shape) == 3:
+            data = data.permute(0, 2, 1)  # cout,k,k,cin
```
Comment (WeiweiZhang1) [inline neural_compressor/compression/pruner/patterns/ninm.py:156]: need to refine note -># cout,k,cin

File: neural_compressor/compression/pruner/patterns/ninm.py [hunk 2]
```diff
@@ -165,7 +168,10 @@ def _reshape_2dims_to_orig(self, data, orig_shape):
         """
         if len(orig_shape) == 4:
             data = data.reshape(orig_shape[0], orig_shape[2], orig_shape[3], orig_shape[1])
-            data = data.permute(0, 3, 1, 2)
```
Comment (WeiweiZhang1) [inline neural_compressor/compression/pruner/patterns/ninm.py:168]: Is this a mistake to overwrite the previous permute?

File: neural_compressor/compression/pruner/patterns/ninm.py (hunk 1 above)
Comment (n1ck-guo) [reply neural_compressor/compression/pruner/patterns/ninm.py]: Done

File: neural_compressor/compression/pruner/patterns/ninm.py (hunk 2 above)
Comment (n1ck-guo) [reply neural_compressor/compression/pruner/patterns/ninm.py:168]: Thanks! Changed

**Review by WeiweiZhang1 — APPROVED** [review]

**Review by wenhuach21 — APPROVED** [review]

**Pull request merged**